src/dnmd/bytes.c

```diff
@@ -159,6 +162,75 @@ bool read_i64(uint8_t const** data, size_t* data_len, int64_t* o)
     return read_le(data, data_len, o, sizeof(*o));
 }
 
+#else
+bool read_u16(uint8_t const** data, size_t* data_len, uint16_t* o)
+{
+    if (*data_len < 2)
+        return false;
+
+    memcpy(o, *data, 2);
+    *data += 2;
+    *data_len -= 2;
```

Comment:
```suggestion
    if (*data_len < sizeof(*o))
        return false;

    memcpy(o, *data, sizeof(*o));
    *data += sizeof(*o);
    *data_len -= sizeof(*o);
```

Using the above, should make all of the basically copy/paste.